src/Simplify.elm

```diff
@@ -10359,26 +10352,52 @@ Use `returnsArgError` with the given last arg as `arg` when the last arg is alre
 alwaysReturnsLastArgError :
     String
     ->
-        { lastArgRepresents : String
-        , lastArg : Maybe (Node Expression)
+        { remainingArgCount : Int
+        , remainingArgs : List (Node Expression)
+        , lastArgRepresents : String
         }
     -> QualifyResources { a | fnRange : Range, parentRange : Range }
     -> Error {}
 alwaysReturnsLastArgError usingSpecificSituation config resources =
-    case config.lastArg of
+    case List.drop (config.remainingArgCount - 1) config.remainingArgs |> List.head of
+        Just lastArg ->
+            returnsArgError usingSpecificSituation { arg = lastArg, argRepresents = config.lastArgRepresents } resources
+
         Nothing ->
+            -- Not enough arguments
+            let
+                replacement : { description : String, fix : List Fix }
+                replacement =
+                    case config.remainingArgCount - List.length config.remainingArgs of
+                        1 ->
+                            { description = "identity"
+                            , fix =
+                                [ Fix.replaceRangeBy resources.parentRange
+                                    (qualifiedToString (qualify ( [ "Basics" ], "identity" ) resources))
+                                ]
+                            }
+
+                        2 ->
+                            { description = "always"
+                            , fix =
+                                [ Fix.replaceRangeBy resources.parentRange
```

Comment:
should be `always identity`

Reply:
```elm
> Array.set -1
<function> : a -> Array.Array a -> Array.Array a
> always identity
<function> : b -> a -> a
```
but
```elm
> always
<function> : a -> b -> a
```
does not unify (the element type and array type aren't necessarily the same)

Reply:
I'll use `always identity` for now, we can always revisit this later.